openmdao/components/exec_comp.py

```diff
@@ -477,9 +482,10 @@ def compute_partials(self, inputs, partials):
                     self._residuals.set_val(0.0)
                     self.compute(pwrap, uwrap)
 
-                    for u in out_names:
-                        # set the column in the Jacobian entry
-                        partials[(u, input)][:, i] = imag(uwrap[u] * inv_stepsize).flat
+                    for out, inp in self._declared_partials.keys():
```

Comment:
I forgot to mention this loop, which has the same issue as the one I mentioned earlier.

Reply:
Done